src/utils/emotes.js

```diff
@@ -0,0 +1,22 @@
+import {emotesCategoryIds} from '../constants.js';
+
+/* eslint-disable import/prefer-default-export */
+export function getEmoteIdFromProvider(emoteId, providerId) {
+  switch (true) {
+    case providerId.startsWith('emoji'):
+    case providerId.startsWith('bttv'):
+      return `${emotesCategoryIds.BETTERTTV}-${emoteId}`;
+    case providerId.startsWith('ffz'):
+      return `${emotesCategoryIds.FRANKERFACEZ}-${emoteId}`;
+    default:
+  }
+
+  switch (providerId) {
+    case emotesCategoryIds.FAVORITES:
```

Comment:
since favorites and frecents are not true emote providers, we may want to throw an error instead and prevent ids from being generated for them